File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpContainerData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpContainerData(
        String dim,
        RMcpBlockPosData pos,
        String side,
        int slots,
        List<Slot> items
) {
    public record Slot(
            int slot,
            String id,
            int count,
            int limit,
            boolean canInsert,
            String snbt
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpContainerData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpContainerData(
    val dim: String,
    val pos: RBlockPos,
    val side: String,
    val slots: Int,
    val items: List<Slot>
) {
    @JvmRecord
    data class Slot(
        val slot: Int,
        val id: String,
        val count: Int,
        val limit: Int,
        val canInsert: Boolean,
        val snbt: String?
    )
}
